src/utils/fs.rs

```diff
@@ -37,22 +37,23 @@ pub fn get_all_files<A: AsRef<Path>>(
         walker.overrides(overrides.build().expect("Excludes provided were invalid"));
     }
 
-    // Flip the booleans as ignore's semantics are the opposite of our options.
-    let no_ignore_vcs = config.no_ignore_vcs.map(|b| !b).unwrap_or(true);
+    let ignore = config.no_ignore.map(|b| !b).unwrap_or(true);
+    let ignore_dot = ignore && config.no_ignore_dot.map(|b| !b).unwrap_or(true);
+    let ignore_vcs = ignore && config.no_ignore_vcs.map(|b| !b).unwrap_or(true);
 
     // Custom ignore files always work even if the `ignore` option is false,
     // so we only add if that option is not present.
-    if !config.no_ignore.unwrap_or(false) {
+    if ignore_dot {
         walker.add_custom_ignore_filename(IGNORE_FILE);
     }
 
     walker
-        .git_exclude(no_ignore_vcs)
-        .git_global(no_ignore_vcs)
-        .git_ignore(no_ignore_vcs)
-        .hidden(config.hidden.map(|b| !b).unwrap_or(true))
-        .ignore(config.no_ignore.map(|b| !b).unwrap_or(true))
-        .parents(config.no_ignore_parent.map(|b| !b).unwrap_or(true));
+        .git_exclude(ignore_vcs)
+        .git_global(ignore_vcs)
+        .git_ignore(ignore_vcs)
+        .hidden(ignore && config.hidden.map(|b| !b).unwrap_or(true))
```

Comment:
Seems like `--no-ignore` would also imply hidden right now. Don't think that should be the case. This should be better:

```suggestion
        .hidden(config.hidden.map(|b| !b).unwrap_or(true))
```

I'll take a closer look and add a test tomorrow/later.